object-construction-checker/src/main/java/org/checkerframework/checker/objectconstruction/MustCallInvokedChecker.java

```diff
@@ -573,8 +573,7 @@ private void handleSuccessorBlocks(
             || setAssign.stream()
                 .allMatch(assign -> succRegularStore.getValue(assign.localVar) == null)) {
           if (nodes.size() == 0) { // If the cur block is special or conditional block
```

Comment:
I think we should only enter this condition when the current block is a conditional block.  Could we add an assertion to this effect?  We should not be doing analysis within special blocks, I think.

Reply:
Adding the following here immediately causes our tests to fail:

```
            if (block.getType() != BlockType.CONDITIONAL_BLOCK) {
              throw new BugInCF("unexpected non-conditional block: " + block.getType());
            }
```

The throw exception looks like this:

```
    :-1: other: error: unexpected non-conditional block: SPECIAL_BLOCK
      ; The Checker Framework crashed.  Please report the crash.
      Compilation unit: /Users/kelloggm/Research/object-construction-checker/object-construction-checker/tests/mustcall/ACOwning.java
      Last visited tree at line 5 column 1:
      class ACOwning {
      Exception: java.lang.Throwable; java.lang.Throwable
        at org.checkerframework.javacutil.BugInCF.<init>(BugInCF.java:16)
        at org.checkerframework.checker.objectconstruction.MustCallInvokedChecker.handleSuccessorBlocks(MustCallInvokedChecker.java:578)
...
```

Reply:
(There is also at least one crash in the `socket` tests.)

Reply:
Ok, let's leave that assertion out then.  I guess we do propagate to special blocks.